tests/TestCase/Mailer/EmailTest.php

```diff
@@ -1875,7 +1875,7 @@ public function testDeliver()
         $this->assertSame($instance->to(), ['debug@cakephp.org' => 'debug@cakephp.org']);
         $this->assertSame($instance->subject(), 'Update ok');
         $this->assertSame($instance->template(), ['template' => 'custom', 'layout' => 'custom_layout']);
-        $this->assertSame($instance->viewVars(), ['value' => 123, 'name' => 'CakePHP']);
+        $this->assertEquals($instance->viewVars(), ['value' => 123, 'name' => 'CakePHP']);
```

Comment:
Because the order of the keys is not the same. Earlier `Email::viewVars()` used `array_merge` while `ViewBuilder` uses `+` to combine new vars with existing ones.